ming/metadata.py

```diff
@@ -358,7 +359,7 @@ def _ensure_indexes(self):
         try:
             with self._lock:
                 for idx in self.manager.indexes:
-                    collection.ensure_index(idx.index_spec, background=True,
+                    collection.create_index(idx.index_spec, background=True,
```

Comment:
https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#collection-ensure-index-is-removed recommends we should avoid frequent calls to `ensure_index`   Are we okay because of the `self.indexes_ensured` flag?  Seems like it